spec/function-builder.spec.ts

```diff
@@ -64,95 +87,111 @@ describe('FunctionBuilder', () => {
     expect(fn.__trigger.timeout).to.deep.equal('90s');
   });
 
-  it('should allow both region and runtime options to be set', () => {
+  it('should allow both supported region and valid runtime options to be set', () => {
     let fn = functions
-      .region('us-east1')
+      .region('europe-west2')
       .runWith({
         timeoutSeconds: 90,
         memory: '256MB',
       })
       .auth.user()
       .onCreate(user => user);
 
-    expect(fn.__trigger.regions).to.deep.equal(['us-east1']);
+    expect(fn.__trigger.regions).to.deep.equal(['europe-west2']);
     expect(fn.__trigger.availableMemoryMb).to.deep.equal(256);
     expect(fn.__trigger.timeout).to.deep.equal('90s');
   });
 
-  it('should allow both region and runtime options to be set (reverse order)', () => {
+  it('should allow both valid runtime options and supported region to be set in reverse order', () => {
     let fn = functions
       .runWith({
         timeoutSeconds: 90,
         memory: '256MB',
       })
-      .region('us-east1')
+      .region('europe-west1')
       .auth.user()
       .onCreate(user => user);
 
-    expect(fn.__trigger.regions).to.deep.equal(['us-east1']);
+    expect(fn.__trigger.regions).to.deep.equal(['europe-west1']);
     expect(fn.__trigger.availableMemoryMb).to.deep.equal(256);
     expect(fn.__trigger.timeout).to.deep.equal('90s');
   });
 
-  it('should throw an error if user chooses an unsupported memory allocation', () => {
+  it('should fail if valid runtime options but unsupported region are set (reverse order)', () => {
+    expect(() => {
+      functions
+        .runWith({ timeoutSeconds: 90, memory: '256MB' })
+        .region('unsupported');
+    }).to.throw(Error);
```

Comment:
Is there anything specific you can check about this error?

Reply:
Checked for `region`, done.